modules/apps/object/object-service/src/main/java/com/liferay/object/internal/field/business/type/PicklistObjectFieldBusinessType.java

```diff
@@ -174,12 +167,30 @@ public void predefineObjectFieldSettings(
 				continue;
 			}
 
-			ObjectStateFlow objectStateFlow =
+			ObjectStateFlow newObjectStateFlow =
 				objectFieldSetting.getObjectStateFlow();
 
-			_objectStateFlowLocalService.addObjectStateFlow(
-				newObjectField.getUserId(), newObjectField.getObjectFieldId(),
-				objectStateFlow.getObjectStates());
+			if (oldObjectField != null) {
+				ObjectStateFlow oldObjectStateFlow =
+					_objectStateFlowLocalService.
+						fetchObjectFieldObjectStateFlow(
+							oldObjectField.getObjectFieldId());
+
+				_objectStateFlowLocalService.
+					updateObjectStateFlowObjectStateTransitions(
+						newObjectField.getUserId(),
+						oldObjectStateFlow.getObjectStateFlowId(),
+						newObjectStateFlow.getObjectStates());
+
+				_objectStateFlowLocalService.updateDefaultObjectStateFlow(
+					newObjectField, oldObjectField);
```

Comment:
@G-Sa since we are only calling the `updateDefaultObjectStateFlow` here, can we move this call to be inside the `updateObjectStateFlow` method?

Reply:
Since we would need to bring in the information of `oldObjectField` and `newObjectField` inside the `updateObjectStateFlow` method, it doesn't make much sense to do this merge of functions